### DISCUSSION TITLE
Metadata question

### DISCUSSION THREAD

**Discussion by R1ch1e-Lee** [rtl-airband/RTLSDR-Airband#428] [General]
Just started running rtl_airband and wanted to learn more about getting metadata working.  I looked through earlier discussions and issues but didn't see an obvious remedy.  I am getting errors in the log and wanted to share it here.  

rtl_airband[1216]: Failed to add shout metadata
rtl_airband[1216]: Activity on 161.370 MHz
rtl_airband[1216]: Activity on 161.520 MHz
rtl_airband[1216]: Failed to add shout metadata
rtl_airband[1216]: Failed to add shout metadata

I'm using an orange pi zero 3 with ubuntu 22.04 lts with libshout-dev version 2.4.5

Thanks
Richard
KB4YTM

**Comment by R1ch1e-Lee** [comment]
some log information from icecast2

[2023-11-03  08:54:30] INFO admin/admin_handle_request Bad or missing password on mount modification admin request (command: metadata)
[2023-11-03  08:55:19] INFO admin/admin_handle_request Bad or missing password on mount modification admin request (command: metadata)
[2023-11-03  09:00:05] INFO admin/admin_handle_request Bad or missing password on mount modification admin request (command: metadata)
[2023-11-03  09:01:49] INFO admin/admin_handle_request Bad or missing password on mount modification admin request (command: metadata)
[2023-11-03  09:04:53] INFO admin/admin_handle_request Bad or missing password on mount modification admin request (command: metadata)
[2023-11-03  09:08:09] INFO admin/admin_handle_request Bad or missing password on mount modification admin request (command: metadata)
[2023-11-03  09:08:46] INFO admin/admin_handle_request Bad or missing password on mount modification admin request (command: metadata)
[2023-11-03  09:29:03] INFO admin/admin_handle_request Bad or missing password on mount modification admin request (command: metadata)
orangepi@orangepizero3:~$ icecast2 -?
Icecast 2.4.4

**Comment by R1ch1e-Lee** [comment]
access.log

127.0.0.1 - - [03/Nov/2023:09:00:05 -0500] "GET /admin/metadata HTTP/1.1" 401 360 "-" "libshout/2.4.5" 0
127.0.0.1 - - [03/Nov/2023:09:01:49 -0500] "GET /admin/metadata HTTP/1.1" 401 360 "-" "libshout/2.4.5" 0
127.0.0.1 - - [03/Nov/2023:09:04:53 -0500] "GET /admin/metadata HTTP/1.1" 401 360 "-" "libshout/2.4.5" 0
127.0.0.1 - - [03/Nov/2023:09:08:09 -0500] "GET /admin/metadata HTTP/1.1" 401 360 "-" "libshout/2.4.5" 0
127.0.0.1 - - [03/Nov/2023:09:08:46 -0500] "GET /admin/metadata HTTP/1.1" 401 360 "-" "libshout/2.4.5" 0
127.0.0.1 - - [03/Nov/2023:09:29:03 -0500] "GET /admin/metadata HTTP/1.1" 401 360 "-" "libshout/2.4.5" 0

**Comment by R1ch1e-Lee** [comment]
libshout version

libshout3-dev is already the newest version (2.4.5-1build3)

**Comment by charlie-foxtrot** [comment]
do the 

```
rtl_airband[1216]: Failed to add shout metadata
```

errors line up with the

```
INFO admin/admin_handle_request Bad or missing password on mount modification admin request (command: metadata)
```

errors?

If so, check the icecast username / password given to `rtl_airband` and see if it has permissions

**Comment by R1ch1e-Lee** [comment]
Thanks for the reply.   Yes they do line up time wise.  I am passing the correct username / password or the icecast stream wouldn't work?  Is that correct?   the stream to icecast and to broadcastify both work normally but still get those errors and no metadata

**Comment by R1ch1e-Lee** [comment]
Is there a separate username/password for the mount and metadata?

**Comment by charlie-foxtrot** [comment]
I haven't used tags so can't say from experience.

Did you have tags running previously?  There were changes to the use of `libshout` in `v4.2.0`, could you try `v4.1.1` and see if it was working then?

**Comment by ryandrake08** [comment]
I've been playing around with the metadata tag setting, too, and have run into the same issue @R1ch1e-Lee observed. Using the `send_scan_freq_tags` option, rtl-airband will try to set the "song" metadata tag to the scan frequency (or optional label). However, it appears that icecast requires one to use the "admin" username/password to set the metadata. The "source" username/password cannot do it. If this is something configurable on the icecast side, I have not found the configuration yet.

If you don't use the admin username/password, icecast will fail to set the metadata and emit the log message:

`admin/admin_handle_request Bad or missing password on mount modification admin request (command: metadata)`

It is possible that a second set of admin credentials would need to be configured and plumbed through rtl-airband, which, from looking at the source code, would appear to be a large project.

@R1ch1e-Lee I am curious about whether you found a workaround since posting this 2 years ago...

EDIT: This behavior is observed using libshout 2.4.5. I'm going to update my Debian distro to one that ships 2.4.6 and see if this makes a difference. libshout lists a couple of fixes made related to setting metadata in 2.4.6, including one I observed on my system: "Fixed shout_set_metadata*() sometimes returning SHOUTERR_RETRY (#2328)" so it could be that this improves with a more recent libshout.

**Reply by R1ch1e-Lee** [reply]
Hello Ryan. While I had the system at home I wiped and reinstalled the OS and it magically began working so I installed it at the site and it still works today.  I didn't dig any deeper into it since it started working but if you make any headway let us all know.    Thanks

**Reply by ryandrake08** [reply]
After updating to using libshout 2.4.6, the tags are working. So that's all it was.
